File: go/217.go
/**
题目概述:
给定一个整数数组，判断是否存在重复元素。

如果存在一值在数组中出现至少两次，函数返回 true 。如果数组中每个元素都不相同，则返回 false 。

*/
package main

func containsDuplicate(nums []int) bool {
   set:=map[int]struct{}{}
   for _,v:=range nums{
   	if _,ok:=set[v];ok{
   		return true
	}
   	set[v]=struct{}{}
   }
   return false

}

File: php/217.php
<?php

class Solution {

           /**
           * @param Integer[] $nums
           * @return Boolean
           */
          function containsDuplicate($nums) {
             $num_arr = [];
             for($i=0;$i<count($nums);$i++){
                        if(isset($num_arr[$nums[$i]])){
                           return true;
                        }
                        $num_arr[$nums[$i]]=1;
             }
             return false;
          }

}